shared/featureconfig/deprecated_flags.go

```diff
@@ -12,14 +12,8 @@ var (
 		Usage:  deprecatedUsage,
 		Hidden: true,
 	}
-	deprecatedDisableSyncBacktracking = &cli.StringFlag{
```

Comment:
shouldnt we leave them all as deprecated ? Anyone else running with the flag will have a failure in the next update.

Reply:
- I suppose Raul asked the same/similar question in https://github.com/prysmaticlabs/prysm/pull/6400 (please see comments in that PR)
- My understanding is the following:
  - this flag doesn't have any effect anymore
  - ppl running with that flag (why would they run with opt-out flag -- the only reason our feature is broken, **that's the purpose we have out-outs, to help resolve issues if we ship sth broken**), were seeing warning that this will be removed
  - there's no point keeping that flag and displaying the warning anymore
 
If you think that opt-out flags should remain as deprecated for longer, just let me know, I will update the PR.

Reply:
This isn't an opt-out flag though, its an opt-in flag just that it is wrongly named. Also we dont generally remove flags in non-minor/major version updates. We should definitely continue keeping on with the pattern, all deprecated flags(no matter how obscure) should only be completely be removed in minor/major version bumps.  So in this case both flags(enable/disable) should continue being deprecated vs completely removed and risk breaking someone in their update.

Reply:
Makes sense, I'll add both enable and disable flags (and name them properly), so that we can remove them in a major version.